toolkit/tools/scheduler/scheduler.go

```diff
@@ -457,6 +476,53 @@ func updateGraphWithImplicitProvides(res *schedulerutils.BuildResult, pkgGraph *
 	return
 }
 
+// setAssociatedDeltaPaths sets the DeltaPath for all of the request's ancillary nodes (both build and run) to the actual
+// RPM paths. A delta node will normally point at the cached RPM path, but we want to point it at the actual RPM if we built it.
+func setAssociatedDeltaPaths(res *schedulerutils.BuildResult, builtFiles []string, pkgGraph *pkggraph.PkgGraph, graphMutex *sync.RWMutex) (err error) {
+	graphMutex.Lock()
+	defer graphMutex.Unlock()
+
+	if !res.WasDelta {
+		err = fmt.Errorf("setAssociatedDeltaPaths called on non-delta node: %s", res.Node.FriendlyName())
+		return
+	}
+
+	// Build map of basename to full path for built files. This will allow us to find the actual RPM path we built for
+	// any given .rpm file built from our ancillary nodes.
+	builtFileMap := make(map[string]string)
+	for _, builtFile := range builtFiles {
+		// We should not expect to see multiple built files with the same basename
+		_, hasConflict := builtFileMap[filepath.Base(builtFile)]
+		if hasConflict {
+			err = fmt.Errorf("multiple built files with same basename: %s", filepath.Base(builtFile))
+			return
+		}
+		logger.Log.Tracef("Built delta file: %s", builtFile)
+		builtFileMap[filepath.Base(builtFile)] = builtFile
+	}
+
+	// Now we can scan for all the run nodes that use the cached RPM path and update them to the actual RPM path.
+	for _, node := range pkgGraph.AllRunNodes() {
+		// Get base path of the .rpm for the node and find the built file in the map
+		rpmBasePath := filepath.Base(node.RpmPath)
+		builtFile, ok := builtFileMap[rpmBasePath]
+		if ok {
+			// We only care about nodes that are deltas
+			if node.State == pkggraph.StateDelta {
+				logger.Log.Tracef("Found delta run node: %s", node)
```

Comment:
>logger.Log.Tracef("Found delta run node: %s", node) [](http://example.com/codeflow?start=4&length=51)

Do we need this first log, since we have the second one and they both always print together? Could we merge them into one?